lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1089,13 +1089,25 @@ InitKindRequest::evaluate(Evaluator &evaluator, ConstructorDecl *decl) const {
     // If we implement the ability for extensions defined in the same module
     // (or the same file) to add vtable entries, we can re-evaluate this
     // restriction.
-    if (dyn_cast<ClassDecl>(nominal) &&
-        !decl->isSynthesized() && isa<ExtensionDecl>(decl->getDeclContext()) &&
+    if (isa<ClassDecl>(nominal) && !decl->isSynthesized() &&
+        isa<ExtensionDecl>(decl->getDeclContext()) &&
         !(decl->getAttrs().hasAttribute<DynamicReplacementAttr>())) {
-      diags.diagnose(decl->getLoc(), diag::designated_init_in_extension,
-                     nominal->getName())
-        .fixItInsert(decl->getLoc(), "convenience ");
-      return CtorInitializerKind::Convenience;
+      auto classDecl = cast<ClassDecl>(nominal);
+
+      if (classDecl->getForeignClassKind() == ClassDecl::ForeignKind::CFType) {
+        diags.diagnose(decl->getLoc(),
+                       diag::cfclass_designated_init_in_extension,
+                       nominal->getName());
+
+        return CtorInitializerKind::Designated;
+      } else {
+        diags
+            .diagnose(decl->getLoc(), diag::designated_init_in_extension,
```

Comment:
:/ Not sure why clang format does this - I think you can manually move this to the line above and move `diag::designated_init_in_extension` argument in a new line. Then, if you run `git-clang-format` again then it shouldn't re-indent like this.

Reply:
Yep, pretty weird. I run `git-clang-format` after moving the lines and still formatted the code as before. So I ended up doing it manually.